packages/grafana-ui/src/components/Logs/logParser.test.ts

```diff
@@ -0,0 +1,170 @@
+          config: {},
+          values: new ArrayVector(['1659620138401000000_8b1f7688_']),
+        },
+      ],
+    });
+    const fields = getAllFields(logRow);
+    expect(fields.length).toBe(1);
+    expect(fields.find((field) => field.key === 'id')).toBe(undefined);
+  });
+
+  it('should filter out entry field which is shown as the log message', () => {
+    const logRow = createLogRow();
+    logRow.entryFieldIndex = 3;
+    logRow.dataFrame = new MutableDataFrame({
+      refId: 'A',
+      fields: [
+        testStringField,
+        {
+          name: 'labels',
+          type: FieldType.other,
+          config: {},
+          values: new ArrayVector([{ place: 'luna', source: 'data' }]),
+        },
+        {
+          name: 'Time',
+          type: FieldType.time,
+          config: {},
+          values: new ArrayVector([1659620138401]),
+        },
+        {
+          name: 'Line',
+          type: FieldType.string,
+          config: {},
+          values: new ArrayVector([
+            '_entry="log text with ANSI \u001b[31mpart of the text\u001b[0m [616951240]" counter=300 float=NaN label=val3 level=info',
+          ]),
+        },
+      ],
+    });
+    const fields = getAllFields(logRow);
+    expect(fields.find((field) => field.key === 'Line')).toBe(undefined);
+  });
+
+  it('should filter out field with config hidden field', () => {
+    const logRow = createLogRow();
+    const testField = { ...testStringField };
+    testField.config = {
+      custom: {
+        hidden: true,
+      },
+    };
+    logRow.dataFrame = new MutableDataFrame({
+      refId: 'A',
+      fields: [{ ...testField }],
+    });
+    const fields = getAllFields(logRow);
+    expect(fields.length).toBe(0);
+    expect(fields.find((field) => field.key === testField.name)).toBe(undefined);
+  });
+
+  it('should filter out field with null values', () => {
+    const logRow = createLogRow();
+    logRow.dataFrame = new MutableDataFrame({
+      refId: 'A',
+      fields: [{ ...testFieldWithNullValue }],
+    });
+    const fields = getAllFields(logRow);
+    expect(fields.length).toBe(0);
+    expect(fields.find((field) => field.key === testFieldWithNullValue.name)).toBe(undefined);
+  });
+
+  it('should not filter out field with string values', () => {
+    const logRow = createLogRow();
+    logRow.dataFrame = new MutableDataFrame({
+      refId: 'A',
+      fields: [{ ...testStringField }],
+    });
+    const fields = getAllFields(logRow);
+    expect(fields.length).toBe(1);
+    expect(fields.find((field) => field.key === testStringField.name)).not.toBe(undefined);
+  });
+});
+
+const testStringField = {
+  name: 'test_field_string',
+  type: FieldType.string,
+  config: {},
+  values: new ArrayVector(['abc']),
+};
+
+const testFieldWithNullValue = {
+  name: 'test_field_null',
+  type: FieldType.string,
+  config: {},
+  values: new ArrayVector([null]),
+};
+
+const createLogRow = (overrides?: Partial<LogRowModel>): LogRowModel => {
```

Comment:
Would it make sense to move this to a common module? Just checking, because it seems like it could be used in a couple of places.

Reply:
Updated in https://github.com/grafana/grafana/pull/53319/commits/553b72ba2cc08312caf7c9a29352ff3dfc2f2768. I have created mock and used it for all of the tests in  Logs folder.